aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -706,6 +706,84 @@ struct ReductionMulOp {
   __forceinline__ scalar_t identity_cpu() const { return 1; }
 };
 
+void _apply_sparse_csr_linear_solve(
+  const Tensor& A,
+  const Tensor& b,
+  const bool left, 
+  const Tensor& x) {
+#if defined(USE_ROCM) || !defined(USE_CUDSS)
+  TORCH_CHECK(
+      false,
+      "Calling torch.linalg.solve with sparse tensors requires compiling ",
+      "PyTorch with CUDA and not supported in ROCm build.");
+#else
+  TORCH_CHECK(A.is_sparse_csr(), "A must be a CSR matrix");
+  TORCH_CHECK(b.dim() == 1, "b must be a 1D tensor");
+  TORCH_CHECK(A.dtype() == b.dtype(), "A and b must have the same dtype");
+  TORCH_CHECK(left == true, "only left == true is supported by the Sparse CSR backend")
+
+  // Device pointers and scalar shape parameters, matrix properties
+
+  Tensor crow = A.crow_indices();
+  Tensor col = A.col_indices();
+  if (crow.scalar_type() != ScalarType::Int) {
+      crow = crow.to(crow.options().dtype(ScalarType::Int));
+      col = col.to(col.options().dtype(ScalarType::Int));
+  }
+  int*    rowOffsets = crow.data<int>();
+  int*    colIndices = col.data<int>();
+  Tensor values     = A.values();
+  //---------------------------------------------------------------------------------
+  // cuDSS data structures and handle initialization
+  cudssConfig_t             config;
+  cudssMatrix_t             b_mt;
+  cudssMatrix_t             A_mt;
+  cudssMatrix_t             x_mt;
+  cudssData_t cudss_data;
+  cudssHandle_t             handle = at::cuda::getCurrentCudssHandle();
+
+  cudssConfigCreate(&config);
+  // cudssAlgType_t reorder_alg = CUDSS_ALG_3;
+  // cudssConfigSet(config, CUDSS_CONFIG_REORDERING_ALG, &reorder_alg, sizeof(cudssAlgType_t));
+  
+  cudssDataCreate(handle, &cudss_data);
+
+  AT_DISPATCH_FLOATING_TYPES(values.type(), "create_matrix", ([&] {
+    scalar_t* values_ptr = values.data<scalar_t>();
+    scalar_t* b_ptr = b.data<scalar_t>();
+    scalar_t* x_ptr = x.data<scalar_t>(); 
+    auto CUDA_R_TYP = std::is_same<scalar_t, double>::value ? CUDA_R_64F : CUDA_R_32F;  // TODO: better conversion
+    cudssMatrixCreateDn(&b_mt, b.size(0), 1, b.size(0), b_ptr, CUDA_R_TYP, CUDSS_LAYOUT_COL_MAJOR);
+    cudssMatrixCreateDn(&x_mt, x.size(0), 1, x.size(0), x_ptr, CUDA_R_TYP, CUDSS_LAYOUT_COL_MAJOR);
```

Comment:
Column- or row-major layout should be checked from the strides of `b` and `x`. The leading dimension argument shouldn't be `b.size(0)` but `max(1, b.stride(0 or 1 based on layout)` to allow a wider variety of strided inputs.

Reply:
Oh, cuDSS documentation says it only supports column-major layout and the leading dimensions equal to exactly the number of rows.
https://docs.nvidia.com/cuda/cudss/functions.html#cudssmatrixcreatedn